packages/api/src/cli/codegen/language.ts

```diff
@@ -36,6 +36,22 @@ export default abstract class CodeGeneratorLanguage {
     // a `petstore` spec installed on api@4.2.0.
     const info = spec.getDefinition().info;
     this.userAgent = `${identifier}/${info.version} (${PACKAGE_NAME}/${PACKAGE_VERSION})`;
+
+    /**
+     * This check is barbaric but there are a number of issues with how the `transformer` work we
+     * have in `oas` and in `.getParametersAsJSONSchema()` and `.getResponseAsJSONSchema()` that
+     * are fully crashing when attempting to codegen an SDK for an API definition that has a
+     * circular reference.
+     *
+     * In order to get v5 out the door we're not going to support this case initialy.
+     *
+     * @see {@link https://github.com/readmeio/api/issues/549}
+     */
+    if (JSON.stringify(spec.api).includes('"$ref":"#/')) {
```

Comment:
The chances of this experiencing false positives where you have a `$ref` property somewhere that isn't a `$ref` is going to be nonexistent, but this check is barbaric nonetheless.